base_classes/NXserialized.nxdl.xml

```diff
@@ -0,0 +1,74 @@
+# version 3 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+# For further information, see http://www.nexusformat.org
+-->
+<definition xmlns="http://definition.nexusformat.org/nxdl/3.1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" category="base" type="group" name="NXserialized" extends="NXobject" xsi:schemaLocation="http://definition.nexusformat.org/nxdl/3.1 ../nxdl.xsd">
+    <doc>
+         Metadata to a set of pieces of information of a resource that has been serialized.
+         
+         A typical use case is the documentation of the source (file) or database (entry)
+         from which pieces of information have been extracted for consumption in e.g. a
+         research data management system (RDMS). This may be for reasons of enabling
+         services such as providing access to normalized information for which reading
+         again from the resource may not be desired, possibe, or feasible.
+         
+         Possible reasons could be the extraction of specific information for caching,
+         performance reasons, or re-evaluate given pieces of information based on other
+         views and interaction patterns with the data where information has been formatted
+         differently by tools than how these pieces of information were originally
+         serialized.
+    </doc>
+    <field name="type" type="NX_CHAR">
+        <doc>
+             Answers into what resource the information was serialized.
+        </doc>
+        <enumeration>
+            <item value="file"/>
+            <item value="database"/>
+        </enumeration>
+    </field>
+    <field name="path" type="NX_CHAR">
+        <doc>
+             Path to the resource.
+             
+             E.g. the name of a file or its absolute or relative path, or the
+             identifier to a resource in another database.
+        </doc>
+    </field>
+    <field name="checksum" type="NX_CHAR">
+        <doc>
+             Value of the hash that is obtained when running algorithm
+             on the content of the resource referred to by path.
+        </doc>
+    </field>
+    <field name="algorithm" type="NX_CHAR">
+        <doc>
+             Name of the algorithm whereby the checksum was computed.
+        </doc>
+    </field>
+    <!--enumeration: [md5, sha256]-->
+    <group name="file" type="NXnote">
+        <doc>
+             Extracted file containing the serialized information.
+        </doc>
+    </group>
+</definition>
```

Comment:
Add the desired parameters to NXnote instead

Reply:
done, removed NXserialized